### PR TITLE
Stray bullets and explosions don't blow up the entire syndi lava base anymore

### LINKED ISSUE #34144: Explosive plastitanium walls destroyed by simple animals will detonate.
This, paired with #34015 means that a plucky goliath situated someplace next to an explosive wall who agros THROUGH the wall can detonate the whole base with no way of seeing this coming.

GENIUS.

### PR DESCRIPTION
**Pull request by DaedalusGame**
:cl: 
balance: you can't get 12 syndi bomb cores from stripping the syndi lava base for parts anymore
fix: fixed the syndi lava base blowing up from stray colossus shots, goliaths breaking in or miners throwing gibtonite at it.
/:cl:

Fixes #34144

Apparently this happens sometimes. Personally I think it's funny but on the other hand many of these explosive walls are placed on the outside walls and it's a bit messy. Also ghost roles get mad about I ded.
Oh yeah, this is also pretty snowflakey. If this has any real chance of getting merged I can make getting the explosion source a helper proc in a proper place.

### CODE REVIEW

File: code/game/turfs/simulated/wall/mineral_walls.dm [hunk 1]
```diff
@@ -250,10 +250,18 @@
...
 	icon_state = "map-overspace"
 	fixed_underlay = list("space"=1)
-
-/turf/closed/wall/mineral/plastitanium/explosive/dismantle_wall(devastated, explode)
-	var/obj/item/bombcore/large/bombcore = new(get_turf(src))
-	if(devastated || explode)
+	
+/turf/closed/wall/mineral/plastitanium/explosive/ex_act(severity)
+	var/datum/explosion/acted_explosion = null
+	log_game("Explosive wall exploded")
```
Comment (JJRcop) [inline code/game/turfs/simulated/wall/mineral_walls.dm]: add ```COORD(src)```

Comment (DaedalusGame) [reply code/game/turfs/simulated/wall/mineral_walls.dm]: fuck that shouldn't be there

**Comment by WJohn** [comment]
Ah, put in fixes #34144 please.

**Comment by DaedalusGame** [comment]
In the changelog part or just the PR?

**Comment by Repukan** [comment]
>miners throwing gibtonite at it.


Seems like a feature tbh

**Comment by DaedalusGame** [comment]
"It's a bunker, it shouldn't explode when people shoot it"

**Comment by Repukan** [comment]
>Throwing explosives = shooting

Gomenasorry I understand now

**Comment by DaedalusGame** [comment]
hey it's not what i said

**Comment by Kromgar** [comment]
@DaedalusGame Were you the syndicate scientist that I popped in on a couple hours ago?

Oh man your reaction in ghost chat was hilarious. The fury and rage. Funny thing was you blew up the base AFTER killing me

**Comment by DaedalusGame** [comment]
No but I think I talked to him on the tg discord, he called me a faggot.

File: code/datums/explosion.dm [hunk 2]
```diff
@@ -14,6 +14,7 @@ GLOBAL_LIST_EMPTY(explosions)
 
 /datum/explosion
 	var/explosion_id
+	var/atom/explosion_source
```
Comment (Cyberboss) [inline code/datums/explosion.dm]: This is gonna cause GC errors because this reference will prevent the source from deleting and it stays forever

**Review by Cyberboss — CHANGES_REQUESTED** [review]

File: code/datums/explosion.dm (hunk 2 above)
Comment (DaedalusGame) [reply code/datums/explosion.dm]: is that right? the explosion datum itself is qdeleted after the explosion finishes, which means that assuming nothing references the explosion datum elsewhere, the explosion datum will GC and the reference to the source will disappear alongside... in a normal programming language...???

Should I null the reference in Destroy anyway?

Comment (Cyberboss) [reply code/datums/explosion.dm]: Nevermind I'm full of shit

**Review by Cyberboss — APPROVED** [review]

**Comment by Cyberboss** [comment]
FYI, save code: in cls for major refactors

**Pull request merged**